src/Shared/StringUtils.cs

```diff
@@ -7,7 +7,7 @@ namespace Aspire.Hosting.Utils;
 
 internal static class StringUtils
 {
-    public static bool TryGetUriFromDelimitedString(string input, string delimiter, [NotNullWhen(true)] out Uri? uri)
+    public static bool TryGetUriFromDelimitedString(string? input, string delimiter, [NotNullWhen(true)] out Uri? uri)
```

Comment:
Can also annotate `input`:

```suggestion
    public static bool TryGetUriFromDelimitedString([NotNullWhen(true)] string? input, string delimiter, [NotNullWhen(true)] out Uri? uri)
```